parley/src/style/mod.rs

```diff
@@ -41,6 +43,42 @@ pub enum OverflowWrap {
     BreakWord,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum LineHeight {
+    /// Line height specified relative to what's defined in the font's metrics.
+    MetricsRelative(f32),
+    /// Line height specified as a multiple of the font size.
+    FontSizeRelative(f32),
+    /// Line height specified in absolute units.
+    Absolute(f32),
+}
+
+impl LineHeight {
+    pub(crate) fn nearly_eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::MetricsRelative(a), Self::MetricsRelative(b))
+            | (Self::FontSizeRelative(a), Self::FontSizeRelative(b))
+            | (Self::Absolute(a), Self::Absolute(b)) => nearly_eq(*a, *b),
+            _ => false,
+        }
+    }
+
+    pub(crate) fn scale(&self, scale: f32) -> Self {
+        match self {
+            Self::Absolute(value) => Self::Absolute(*value * scale),
+            value => *value,
+        }
+    }
+
+    pub(crate) fn resolve(&self, scale: f32) -> LayoutLineHeight {
+        match self {
+            Self::MetricsRelative(value) => LayoutLineHeight::MetricsRelative(*value),
+            Self::FontSizeRelative(value) => LayoutLineHeight::Absolute(*value * scale),
+            Self::Absolute(value) => LayoutLineHeight::Absolute(*value),
+        }
+    }
```

Comment:
Let's get this back too.

```suggestion
    pub(crate) fn resolve(&self, font_size: f32) -> LayoutLineHeight {
        match self {
            Self::MetricsRelative(value) => LayoutLineHeight::MetricsRelative(*value),
            Self::FontSizeRelative(value) => LayoutLineHeight::Absolute(*value * font_size),
            Self::Absolute(value) => LayoutLineHeight::Absolute(*value),
        }
    }
```